major_event_log/views.py

```diff
@@ -18,32 +28,33 @@ def index(request):
 
 def event_details(request, event_id):
     """Loads the event details page of the event with the given id."""
-    try:
-        event = Event.objects.get(id=event_id)
-    except Event.DoesNotExist:
-        raise Http404
+    if is_uuid(event_id) == True:
+        event = get_object_or_404(Event.objects, id=event_id)
+    else:
+        raise Http404("Invalid event id")
```

Comment:
I think we can abstract this even more. We are still duplicating a lot of the same code.

What if we did something like

```python
def get_event_or_404(id):
    try:
        uuid.UUID(id)
    except ValueError:
        raise Http404
    return get_object_or_404(Event, id=event_id)
```